extension/src/frequency.rs

```diff
@@ -445,6 +601,36 @@ pub fn freq_agg_trans(
         {SpaceSavingTransState::freq_agg_from_type_id(freq, typ, collation)}).internal()
 }
 
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+pub fn freq_agg_bigint_trans(
+    state: Internal,
+    freq: f64,
+    value: Option<i64>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal> {
+    let value = match value {
+        None => None,
+        Some(val) => unsafe {AnyElement::from_datum(val as pg_sys::Datum, false, pg_sys::INT8OID)}
+    };
+    freq_agg_trans(state, freq, value, fcinfo)
+}
+
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+pub fn freq_agg_text_trans(
+    state: Internal,
+    freq: f64,
+    value: Option<&str>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal> {
+    // Important: this variable will hold the lifetime of the varlena until we copy it in freq_agg_trans()
+    let text = value.map(pgx::rust_str_to_text_p);
```

Comment:
if we're doing this I'd recommend dust using an `Option<crate::raw::text>` and using `pg_detoast_datum_copy()`